packages/flutter/lib/src/material/typography.dart

```diff
@@ -6,623 +6,449 @@ import 'package:flutter/foundation.dart';
+  final TextTheme englishLike;
+
+  /// Defines text geometry for dense scripts, such as Chinese, Japanese
+  /// and Korean.
   ///
-  ///   final Color titleColor;
-  ///   final Widget child;
+  /// This text theme is merged with either [black] or [white], depending
+  /// on the overall [ThemeData.brightness], when the current locale's
+  /// [MaterialLocalizations.scriptCategory] is [ScriptCategory.dense].
   ///
-  ///   @override
-  ///   Widget build(BuildContext context) {
-  ///     ThemeData theme = Theme.of(context);
-  ///     // This partialTheme is incomplete: it only has the title style
-  ///     // defined. Just replacing theme.textTheme with partialTheme would
-  ///     // set the title, but everything else would be null. This isn't very
-  ///     // useful, so merge it with the existing theme to keep all of the
-  ///     // preexisting definitions for the other styles.
-  ///     TextTheme partialTheme = TextTheme(title: TextStyle(color: titleColor));
-  ///     theme = theme.copyWith(textTheme: theme.textTheme.merge(partialTheme));
-  ///     return Theme(data: theme, child: child);
-  ///   }
-  /// }
-  /// ```
+  /// To look up a localized [TextTheme], use the the overall [Theme], for
+  /// example: `Theme.of(context).textTheme`.
+  final TextTheme dense;
+
+  /// Defines text geometry for tall scripts, such as Farsi, Hindi, and Thai.
   ///
-  /// See also:
+  /// This text theme is merged with either [black] or [white], depending
+  /// on the overall [ThemeData.brightness], when the current locale's
+  /// [MaterialLocalizations.scriptCategory] is [ScriptCategory.tall].
   ///
-  ///   * [copyWith] is used instead of [merge] when you wish to override
-  ///     individual fields in the [TextTheme] instead of merging all of the
-  ///     fields of two [TextTheme]s.
-  TextTheme merge(TextTheme other) {
-    if (other == null)
-      return this;
-    return copyWith(
-      display4: display4?.merge(other.display4) ?? other.display4,
-      display3: display3?.merge(other.display3) ?? other.display3,
-      display2: display2?.merge(other.display2) ?? other.display2,
-      display1: display1?.merge(other.display1) ?? other.display1,
-      headline: headline?.merge(other.headline) ?? other.headline,
-      title: title?.merge(other.title) ?? other.title,
-      subhead: subhead?.merge(other.subhead) ?? other.subhead,
-      body2: body2?.merge(other.body2) ?? other.body2,
-      body1: body1?.merge(other.body1) ?? other.body1,
-      caption: caption?.merge(other.caption) ?? other.caption,
-      button: button?.merge(other.button) ?? other.button,
-    );
+  /// To look up a localized [TextTheme], use the the overall [Theme], for
+  /// example: `Theme.of(context).textTheme`.
+  final TextTheme tall;
+
+  /// Returns one of [englishLike], [dense], or [tall].
+  TextTheme geometryThemeFor(ScriptCategory category) {
+    assert(category != null);
+    switch(category) {
```

Comment:
nit: switch isn't a method call, so there should be a space after the keyword

Reply:
cc @a14n if you would like an easy lint to add... `if(`, `while(`, `for(`, `switch(` -- none of these are method calls, so all should have a space between the keyword and the expression to distinguish them visually from method calls.

Reply:
Done